torch/quantization/quantize.py

```diff
@@ -76,7 +81,7 @@ def add_observer(module):
     if hasattr(module, 'qconfig') and module.qconfig is not None and len(module._modules) == 0:
         # observer and hook will be gone after we swap the module
         module.add_module('observer', module.qconfig.activation())
-        module.register_forward_hook(_observer_forward_hook)
+        module.register_forward_hook(_fake_quant_forward_hook if module.training else _observer_forward_hook)
```

Comment:
@dzhulgakov btw we still rely on module.training here, let me know if you want me to change this

Reply:
@jerryzh168 - I don't think you need it. You can just have a single hook that does `return observer(output)`. In case of actual observer, it returns None so the output won't be overwritten. Effectively we pass down the contract of forward hooks down to observer (we should document it). Want to fix it?

Reply:
sounds hacky, should we do that?

Reply:
I'd say it's less magical than checking .training here. Btw, if .training is wrong, e.g. it gets to be 'true' for PQT, you'd do exactly what I've suggested :)

Reply:
we are putting a constraint on observer that it will always return None in forward and on FakeQuant module that it will return a valid output in forward though. 

The constraint for observer will be relaxed in current implementation, but we require people to get the training flag right, which is another constraint. when .training is wrong the FakeQuant module will just be an observer. which still make sense actually.

I don't have much preference in this case. we can use the original one.